src/main/java/core/basesyntax/strategyimpl/PurchaseHandler.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.strategyimpl;
+
+import core.basesyntax.storage.Storage;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseHandler implements OperationHandler {
+    @Override
+    public void calculateValue(String fruit, int quantity) {
+        if (Storage.fruits.containsKey(fruit)) {
+            Storage.fruits.put(fruit, Storage.fruits.get(fruit) - quantity);
+        }
```

Comment:
I didnt saw all files but how you will handle the situation when you want to try to perchase fruits that you dont have like there where no supply or balance of such fruits 
IMHO - exception needed - remainder i didnt read all files yet so if you got this check later than ok